packages/composites/circuit-breaker/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,191 @@
+import { assertError } from '@chainlink/ea-test-helpers'
+import { Requester } from '@chainlink/ea-bootstrap'
+import * as circuitbreakerAllocationAdapter from '../../src/index'
+import { dataProviderConfig, mockDataProviderResponses } from './fixtures'
+import nock from 'nock'
+
+let oldEnv: NodeJS.ProcessEnv
+
+describe('execute', () => {
+  let execute: Execute
+
+  beforeAll(async () => {
+    execute = await circuitbreakerAllocationAdapter.makeExecute()
+    if (process.env.RECORD) {
+      nock.recorder.rec()
+    }
+  })
+
+  afterAll(() => {
+    process.env = oldEnv
+    if (process.env.RECORD) {
+      nock.recorder.play()
+    }
+
+    nock.restore()
+    nock.cleanAll()
+    nock.enableNetConnect()
+  })
+
+  describe('valid result method', () => {
+    oldEnv = JSON.parse(JSON.stringify(process.env))
+    for (const source of Object.keys(dataProviderConfig)) {
+      const { providerUrlEnvVar, providerUrl } = dataProviderConfig[source]
+      process.env[providerUrlEnvVar] = providerUrl
+    }
+
+    mockDataProviderResponses()
+    const jobID = '1'
+    const request = [
+      {
+        name: 'should return correct result value with the second source if the first one fail',
+        input: {
+          id: jobID,
+          data: {
+            primarySource: 'none',
```

Comment:
Does this source simply not exist/isn't configured - or does it return a 500/error response?

We should test with the adapter returning a response, but with a 500 status code.

Reply:
none is for testing ECONNREFUSED